lib/output_helper.rb

```diff
@@ -336,14 +336,8 @@ def self.generate_partitions_geometry(result)
       return unless has_vehicle_partition || has_work_day_partition
 
       partitions = {}
-      if has_vehicle_partition
-        partitions[:vehicle] = draw_cluster(elements, :vehicle)
-      end
-
-      return partitions unless has_work_day_partition
-
-      partitions[:work_day] = draw_cluster(elements, :work_day)
-
+      partitions[:vehicle] = draw_cluster(elements, :vehicle) if has_vehicle_partition
+      partitions[:work_day] = draw_cluster(elements, :work_day) if has_work_day_partition
```

Comment:
We can even move the `if` conditions into the `draw_cluster` function so that it manages its own extreme cases itself, no? Like "`return nil if entity == :vehicle && has_NO_vehicle_partition`" but then we need to parse the elements twice :/